apps/ex_wire/lib/ex_wire/struct/warp_queue.ex

```diff
@@ -0,0 +1,275 @@
+defmodule ExWire.Struct.WarpQueue do
+  @moduledoc """
+  `WarpQueue` maintains the current state of an active warp, this mean we will 
+  track the block_chunk hashes and state_chunk hashes given to us, so we can
+  request each from peers.
+
+  TODO: This will likely need to be updated to handle warping from more than
+        one direct peer.
+  """
+
+  alias Block.Header
+  alias Blockchain.{Block, Blocktree}
+  alias ExWire.Packet.SnapshotData.BlockChunk
+  alias ExWire.Packet.SnapshotManifest
+  alias MerklePatriciaTree.Trie
+
+  @type t :: %{
+          manifests: [SnapshotManifest.t()],
+          first_block_hash: EVM.hash() | nil,
+          current_block_number: integer() | nil,
+          current_block_hash: EVM.hash() | nil,
+          known_block_hashes: MapSet.t(EVM.hash()),
+          known_stash_hashes: MapSet.t(EVM.hash()),
+          block_chunk_requests: MapSet.t(EVM.hash()),
+          unprocessed_block_data: %{integer() => BlockData.t()}
+        }
+  defstruct [
+    :manifests,
+    :first_block_hash,
+    :current_block_number,
+    :current_block_hash,
+    :known_block_hashes,
+    :known_state_hashes,
+    :block_chunk_requests,
+    :unprocessed_block_data
+  ]
+
+  @doc """
+  Creates a new `WarpQueue` that 
```

Comment:
Does this need to be expanded? Otherwise we could lose the `that`:
```suggestion
  Creates a new `WarpQueue`.
```